sponsor-dapp-v2/src/lib/custom-hooks.js

```diff
@@ -62,8 +62,8 @@ export function useFaucetUrls() {
 export function useTextInput() {
   const [amount, setAmount] = useState("");
   const handleChangeAmount = event => {
-    // Check if regex number matches
-    if (/^(\s*|\d+)$/.test(event.target.value)) {
+    // Regular expression that matches a decimal, e.g., `2.5`.
+    if (/^(\d+\.?\d*)$/.test(event.target.value)) {
```

Comment:
nit: any reason why we shouldn't do `d*\.?\d*` to allow the .2 case? Is there some unintended consequence or something I'm missing?

Reply:
I haven't done regexes in a while, so forgive me if this is obviously wrong :)

Reply:
I think that would allow both `""` (empty string) and `"."` (just a dot), which would fail to parse later on. I haven't done regexes in a while either, so I'm happy to go with whatever regex you suggest.

Reply:
Added a special case for empty string.